### PR TITLE
Allow FTRL training for a fractional number of epochs

### LINKED ISSUE #2263: FTRL model should allow training for a fractional number of epochs
The type of the FTRL `nepochs` parameter should be changed from `int` to `float` allowing training for a fractional number of epochs. 

Since early stopping in `.fit()` can be triggered in multiples of  `nepochs_validation` and this parameter has a `float` type — there is no reason to prevent training with no validation for explicitly specified fractional number of epochs .

### PR DESCRIPTION
**Pull request by oleksiyskononenko**
Closes #2263

### CODE REVIEW

File: c/models/dt_ftrl.cc [hunk 1]
```diff
@@ -556,16 +558,20 @@ FtrlFitOutput Ftrl<T>::fit(T(*linkfn)(T),
 
   // Training settings. By default each training iteration consists of
   // `dt_X_train->nrows` rows.
-  size_t niterations = params.nepochs;
+  size_t niterations = static_cast<size_t>(std::ceil(nepochs));
+  T frac_nepochs = nepochs - niterations + 1;
```
Comment (st-pasha) [inline c/models/dt_ftrl.cc]: if `nepochs` is integer, say `3`, then `niterations = 3` too, and then `frac_nepochs = 1`. Is this what you intended?

File: c/models/py_ftrl.cc [hunk 2]
```diff
@@ -1200,7 +1202,7 @@ mantissa_nbits : int
     Number of bits from mantissa to be used for hashing floats,
     defaults to `10`.
 
-nepochs : int
+nepochs : float
     Number of training epochs, defaults to `1`.
```
Comment (st-pasha) [inline c/models/py_ftrl.cc]: Perhaps add an explanation here what does it mean for `nepochs` to be fractional

File: c/models/dt_ftrl.cc (hunk 1 above)
Comment (oleksiyskononenko) [reply c/models/dt_ftrl.cc]: Yeah, but the variable names I've used here could really cause some confusion. I've streamlined this part and added more comments.

File: c/models/py_ftrl.cc (hunk 2 above)
Comment (oleksiyskononenko) [reply c/models/py_ftrl.cc]: Sure, done.

**Pull request merged**